src/main/webapp/app/exercises/shared/exercise/exercise-details/exercise-details.component.html

```diff
@@ -28,8 +28,8 @@
 </ng-container>
 <dt><span jhiTranslate="artemisApp.exercise.categories">Difficulty</span></dt>
 <dd *ngIf="exercise.categories">
-    <span class="custom-tag" *ngFor="let item of exercise.categories" [ngStyle]="{ backgroundColor: item.color }">
-        <span>{{ item.category }}</span>
+    <span class="custom-tag" *ngFor="let item of exercise.categories" [ngStyle]="{ backgroundColor: item }">
```

Comment:
this change seems wrong to me. Please note that categories are stored in the database like this:

`{"category":"test","color":"#0ab84f"}`

They are mapped like this, e.g. in `programming-exercise.detail.component.ts`

`this.programmingExercise.categories = this.programmingExercise.categories.map((category) => JSON.parse(category));`

A proper solution would be to define a custom class in the client for categories with a string category attribute and a string color attribute. Then this could also be more type safe.

Could you please change that?